### PR TITLE
Add X-Remaining-Downloads and X-Remaining-Days headers to HEAD and GET responses

### PR DESCRIPTION
**Pull request by dopessoa**
This commit satisfies the request on issue #229.

### CODE REVIEW

File: server/handlers.go [hunk 1]
```diff
@@ -583,6 +583,35 @@ func (s *Server) Unlock(token, filename string) error {
 	return nil
 }
 
+func (s *Server) GetMetadata(token, filename string) (Metadata, error) {
```
Comment (aspacca) [inline server/handlers.go]: please, don't add a 99% identical method (see `CheckMetadata`), just refactor the existing one to return `Metadata`

also related to head requests increasing the download count `CheckMetadata` can become:
`func (s *Server) ProcessMetadata(token, filename string, increaseDownload bool) (Metadata, error) {`

File: server/handlers.go [hunk 2]
```diff
@@ -870,6 +899,8 @@ func (s *Server) headHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	metadata, err := s.GetMetadata(token, filename)
```
Comment (aspacca) [inline server/handlers.go]: see first comment: we are doing basically two times the same thing in https://github.com/dutchcoders/transfer.sh/pull/234/files#diff-105fedf355a9fcad82982745cef24f7aR896 and here

File: server/handlers.go [hunk 3]
```diff
@@ -918,10 +957,16 @@ func (s *Server) getHandler(w http.ResponseWriter, r *http.Request) {
 		disposition = "attachment"
 	}
 
+	remainingDownloads := metadata.MaxDownloads - metadata.Downloads
```
Comment (aspacca) [inline server/handlers.go]: please, extract these three repeated lines from here https://github.com/dutchcoders/transfer.sh/pull/234/files#diff-105fedf355a9fcad82982745cef24f7aR914 in separate methods of `MetaData` encapsulating the logic

**Comment by dopessoa** [comment]
Thank you for the review, I've made some changes. It's much better now.

This PR fixes #235 as well. Should I separate the commits?

**Comment by aspacca** [comment]
thanks @dopessoa , looks good to me. no need to create two separate PR
can you check the failure in travis?

**Pull request merged**